postrelease-vip.php

```diff
@@ -2,567 +2,680 @@
-	if( ! $page_id ) {
-		postrelease_create_page();
-		return false;
-	}
-	
-	$page = get_page($page_id);
-	
-	if($page == null || 
-	   strcmp($page->post_status,'publish') != 0 || 
-	   strstr($page->post_name, 'postrelease') == false ||
-	   strcmp($page->post_type, PRX_CUSTOM_TYPE) != 0 ||	   
-	   strcmp($page->post_title,'<span class="prx_title"></span>') != 0 || 
-	   strcmp($page->post_content,'<span class="prx_body"></span>') != 0
-	   ) 
-	{
-		postrelease_create_page();
-		return false;
-	}
-	
-	return true;
-}
+		}
 
-/**
- * Allow additional parameters in the URL
- * @param unknown_type $vars
- */
-function postrelease_add_query_vars($vars) {
-	$vars[] = "prx_t";
-	$vars[] = "prx_rk";
-	$vars[] = "prx_ro";
-	$vars[] = "prx";
-	return $vars;
-}
-
-/**
- * Create the ad template post 
- */
-function postrelease_create_page() {
-    $current_user = wp_get_current_user();
-    
-	$template_post = array(
-	    		'post_title' => '<span class="prx_title"></span>',
-	    		'post_content' => '<span class="prx_body"></span>',
-	    		'post_type' => PRX_CUSTOM_TYPE,
-	    		'post_status' => 'publish',
-	    		'post_author' => $current_user->ID,
-	    		'post_name' => 'postrelease',
-				'post_date' => "1960-01-01 00:0:00",
-				'post_date_gmt' => "1960-01-01 00:00:00",
-	    		'comment_status' => 'closed',
-				'ping_status' => 'closed',
-    		);
-
-    $template_post_id = wp_insert_post($template_post);
-
-    //save in wordpress db so we can remove that when deactivating plugin
-    update_option('prx_template_post_id', $template_post_id);
-}
-
-
-function postrelease_template_redirect() {
-	global $wp_query;
-
-	// functions that don't require security check
-	if(isset($wp_query->query_vars['prx'])) { //prx is a URL parameter
-		$function = $wp_query->query_vars['prx'];	
-		
-		if ($function == 'ad') {	// redirect to full ad page		
-			if(get_option('prx_plugin_activated', 0) == 1) {
-
-				//redirect to template page
-				$template_post_id = get_option('prx_template_post_id', 1);
-
-				if ( ! $template_post_id )
-					return;
-
-				$template_post_url = get_permalink($template_post_id);
-
-				$query_struct = array();
-				parse_str($_SERVER['QUERY_STRING'], $query_struct);
-				$template_post_url = add_query_arg( $query_struct, $template_post_url ); //keeping URL parameters when we redirect (prx_t and prx_rk need to stay in the URL)				
-				$template_post_url = add_query_arg( 'prx', 'page', $template_post_url ); //prx=page so that we do not get redirected anymore
-				wp_redirect( $template_post_url );
-				
-				exit;
+		/**
+		 * Appends the PostRelease javascript to header
+		 *
+		 * @return void
+		 */
+		public function postrelease_wp_enqueue_scripts() {
+			wp_enqueue_script( 'postrelease', $this->postrelease_js_url, [], null, false );
+		}
+
+		/**
+		 * Register postrelease post type
+		 *
+		 * @return void
+		 */
+		public function postrelease_init() {
+			$args = [
+				'public' => false || $this->is_dev,
```

Comment:
This comes up a couple times, but the `false ||` is pointless. This will always be the value of `$this->is_dev`